### PR TITLE
Remove DeploymentID from response headers

### PR DESCRIPTION
**Pull request by kannappanr**
Response headers need not contain deployment ID.

## Description

MinIO responds with a header named `X-Minio-Deployment-ID`, so when we do a copy from a MinIO source to any other S3 server, the header `X-Minio-Deployment-ID` becomes `X-AMZ-X-Minio-Deployment-ID`. We need not send the header `X-Minio-Deployment-Id` as a response header.

## Motivation and Context
Desribed in the description. 

## Regression
No

## How Has This Been Tested?

```
mc cp /etc/issue myminio/bucket/object
mc stat play/test/object
```

The output should not contain a header called `X-Minio-deployment-Id` like in the following output
```
Name      : object
Date      : 2019-06-25 13:30:30 PDT 
Size      : 55 MiB 
ETag      : 4291fae1f746c6bafd4a9013da08d2f5 
Type      : file 
Metadata  :
  Content-Type         : application/x-executable 
  X-Minio-Deployment-Id: 41e39f4a-3b66-415b-9ddf-025d76a58668 
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :x: [more...](#a0a95ea7-70b0-4824-a0e9-865eae6de1e9) |
| mint-dist-xl.sh | :x: [more...](#088e16c4-af07-4abd-bfe2-fd27396a46da) |
<a name="a0a95ea7-70b0-4824-a0e9-865eae6de1e9"></a>
### 7815-077b060/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:31289
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 18543fbc94b5:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-25bi0jdr9ibrav",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3072,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:31289/aws-sdk-go-test-25bi0jdr9ibrav: dial tcp 72.28.97.57:31289: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="088e16c4-af07-4abd-bfe2-fd27396a46da"></a>
### 7815-077b060/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:30361
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp aea6ded9dad5:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 59 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 56 seconds
(8/14) Running minio-dotnet tests ... done in 37 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 0 seconds
(10/14) Running minio-java tests ... FAILED in 49 seconds
{
  "name": "minio-java",
  "function": "presignedGetObject(String bucketName, String objectName, Integer expires)",
  "duration": 37,
  "status": "FAIL",
  "error": "java.lang.Exception: failed to read object. Response: Response{protocol=http/1.1, code=503, message=Service Unavailable, url=http://72.28.97.61:30361/minio-java-test-2rnalh/minio-java-test-3tn6pdv?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20190621%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20190621T014932Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=07fc9d9133a7c39bb03b94278907bc50740268d6acbd4220189782ba53ed7d67}, Response body: <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Please reduce your request</Message><Key>minio-java-test-3tn6pdv</Key><BucketName>minio-java-test-2rnalh</BucketName><Resource>/minio-java-test-2rnalh/minio-java-test-3tn6pdv</Resource><RequestId>15AA1399C5BC35B5</RequestId><HostId>793f17bb-6b9a-41ed-afa1-ea3dc7cf79e3</HostId></Error> >>> [FunctionalTest.readObject(FunctionalTest.java:201), FunctionalTest.presignedGetObject_test2(FunctionalTest.java:1803), FunctionalTest.runTests(FunctionalTest.java:2995), FunctionalTest.main(FunctionalTest.java:3118)]"
}

Executed 9 out of 14 tests successfully.
```
{"name": "7815-077b060", "full_size": 63679169, "images": [{"size": 63679169, "architecture": "amd64", "variant": null, "features": null, "os": "linux", "os_version": null, "os_features": null}], "id": 59610642, "repository": 757613, "creator": 21164, "last_updater": 21164, "last_updated": "2019-06-21T01:31:21.243711Z", "image_id": null, "v2": true}

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#a7472b5d-cb37-480a-a318-6e895262c1d5) |
| mint-compression-fs.sh | :x: [more...](#f9630d2f-7234-4a5a-a85e-3bd99d14ae07) |
| mint-dist-xl.sh | :x: [more...](#3e82adee-05c8-45cd-8253-cb5888aa5e38) |
<a name="f9630d2f-7234-4a5a-a85e-3bd99d14ae07"></a>
### 7815-7b271dd/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:31650
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f1cad8064ddc:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 1 seconds
(5/14) Running awscli tests ... done in 1 minutes and 45 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 27 seconds
(8/14) Running minio-dotnet tests ... done in 23 seconds
(9/14) Running minio-go tests ... done in 49 seconds
(10/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "putObject(String bucketName, String objectName, String filename)",
  "args": "filename: 6MB",
  "duration": 532,
  "status": "FAIL",
  "error": "java.io.EOFException >>> [okio.RealBufferedSink.write(RealBufferedSink.java:115), io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:77), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:264), okhttp3.RealCall.execute(RealCall.java:93), io.minio.MinioClient.executeReq(MinioClient.java:1111), io.minio.MinioClient.execute(MinioClient.java:1070), io.minio.MinioClient.executePut(MinioClient.java:1405), io.minio.MinioClient.executePut(MinioClient.java:1427), io.minio.MinioClient.putObject(MinioClient.java:4216), io.minio.MinioClient.putObject(MinioClient.java:4328), io.minio.MinioClient.putObject(MinioClient.java:3591), FunctionalTest.putObject_test2(FunctionalTest.java:440), FunctionalTest.runTests(FunctionalTest.java:2957), FunctionalTest.main(FunctionalTest.java:3118)]"
}

Executed 9 out of 14 tests successfully.
```
<a name="a7472b5d-cb37-480a-a318-6e895262c1d5"></a>
### 7815-7b271dd/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:31007
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0ca270d6dbde:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 47 seconds
(4/14) Running aws-sdk-ruby tests ... done in 16 seconds
(5/14) Running awscli tests ... done in 2 minutes and 14 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... FAILED in 2 minutes and 35 seconds
{
  "name": "mc",
  "duration": "115597",
  "function": "test_mirror_list_objects_storage_class",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-6520 --quiet --no-color mirror --storage-class REDUCED_REDUNDANCY /mint/data myminio/mc-test-bucket-29543  >>> \n`/mint/data/datafile-1-MB` -> `myminio/mc-test-bucket-29543/datafile-1-MB`\n`/mint/data/datafile-0-b` -> `myminio/mc-test-bucket-29543/datafile-0-b`\n`/mint/data/datafile-10-MB` -> `myminio/mc-test-bucket-29543/datafile-10-MB`\n`/mint/data/datafile-1-b` -> `myminio/mc-test-bucket-29543/datafile-1-b`\n`/mint/data/datafile-10-kB` -> `myminio/mc-test-bucket-29543/datafile-10-kB`\n`/mint/data/datafile-11-MB` -> `myminio/mc-test-bucket-29543/datafile-11-MB`\n`/mint/data/datafile-1-kB` -> `myminio/mc-test-bucket-29543/datafile-1-kB`\n`/mint/data/datafile-129-MB` -> `myminio/mc-test-bucket-29543/datafile-129-MB`\n`/mint/data/datafile-5-MB` -> `myminio/mc-test-bucket-29543/datafile-5-MB`\n`/mint/data/datafile-1.03-MB` -> `myminio/mc-test-bucket-29543/datafile-1.03-MB`\n`/mint/data/datafile-100-kB` -> `myminio/mc-test-bucket-29543/datafile-100-kB`\n`/mint/data/datafile-33-kB` -> `myminio/mc-test-bucket-29543/datafile-33-kB`\n`/mint/data/datafile-5243880-b` -> `myminio/mc-test-bucket-29543/datafile-5243880-b`\n`/mint/data/datafile-65-MB` -> `myminio/mc-test-bucket-29543/datafile-65-MB`\n`/mint/data/datafile-6-MB` -> `myminio/mc-test-bucket-29543/datafile-6-MB`\nmc: <ERROR> Failed to copy `/mint/data/datafile-65-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-129-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-11-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-10-MB`. Please reduce your request\nTotal: 233.17 MiB, Transferred: 585.24 MiB, Speed: 5.07 MiB/s"
}

Executed 6 out of 14 tests successfully.
```
<a name="3e82adee-05c8-45cd-8253-cb5888aa5e38"></a>
### 7815-7b271dd/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:31764
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 697a07f36a15:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 58 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 34 seconds
(8/14) Running minio-dotnet tests ... done in 32 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 34 seconds
(10/14) Running minio-java tests ... FAILED in 40 seconds
{
  "name": "minio-java",
  "function": "listIncompleteUploads(String bucketName, String prefix)",
  "duration": 1084,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=SlowDown, message=Please reduce your request, bucketName=minio-java-test-3f56542, objectName=minio-java-test-d6p3nu, resource=/minio-java-test-3f56542/minio-java-test-d6p3nu, requestId=15AA174504068870, hostId=5c39600c-3187-4434-9257-a5c1d4d47d58)\nrequest={method=DELETE, url=http://72.28.97.53:31764/minio-java-test-3f56542/minio-java-test-d6p3nu?uploadId=be620cd8-2233-4011-9464-c9d01423b65f, headers=Host: 72.28.97.53:31764\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20190621T025646Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20190621/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 361\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2019-06-21T02-38-40Z\nVary: Origin\nX-Amz-Request-Id: 15AA174504068870\nX-Xss-Protection: 1; mode=block\nDate: Fri, 21 Jun 2019 02:56:46 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1204), io.minio.MinioClient.execute(MinioClient.java:1070), io.minio.MinioClient.executeDelete(MinioClient.java:1347), io.minio.MinioClient.abortMultipartUpload(MinioClient.java:5149), io.minio.MinioClient.putObject(MinioClient.java:4337), io.minio.MinioClient.putObject(MinioClient.java:4183), FunctionalTest.listIncompleteUploads_test2(FunctionalTest.java:1639), FunctionalTest.runTests(FunctionalTest.java:2989), FunctionalTest.main(FunctionalTest.java:3118)]"
}

Executed 9 out of 14 tests successfully.
```
{"name": "7815-7b271dd", "full_size": 63679182, "images": [{"size": 63679182, "architecture": "amd64", "variant": null, "features": null, "os": "linux", "os_version": null, "os_features": null}], "id": 59615194, "repository": 757613, "creator": 21164, "last_updater": 21164, "last_updated": "2019-06-21T02:39:37.767240Z", "image_id": null, "v2": true}

File: cmd/object-handlers.go [hunk 1]
```diff
@@ -411,7 +412,7 @@ func (api objectAPIHandlers) GetObjectHandler(w http.ResponseWriter, r *http.Req
 // The HEAD operation retrieves metadata from an object without returning the object itself.
 func (api objectAPIHandlers) HeadObjectHandler(w http.ResponseWriter, r *http.Request) {
 	ctx := newContext(r, w, "HeadObject")
-
+	logger.LogIf(ctx, errors.New("Test Error"))
```
Comment (krisis) [inline cmd/object-handlers.go]: nit: looks like a debug statement

**Review by krisis — CHANGES_REQUESTED** [review]

File: cmd/object-handlers.go (hunk 1 above)
Comment (kannappanr) [reply cmd/object-handlers.go]: Yes. Will remove it

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#5bd409b8-9b23-4f49-a4ae-be07c1611dcf) |
<a name="5bd409b8-9b23-4f49-a4ae-be07c1611dcf"></a>
### 7815-d925e1b/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:32333
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 375ff3ac99c1:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 57 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 59 seconds
(8/14) Running minio-dotnet tests ... done in 38 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 1 seconds
(10/14) Running minio-java tests ... FAILED in 39 seconds
{
  "name": "minio-java",
  "function": "listIncompleteUploads(String bucketName)",
  "duration": 605,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=SlowDown, message=Please reduce your request, bucketName=minio-java-test-3r6ra4h, objectName=minio-java-test-2be2djh, resource=/minio-java-test-3r6ra4h/minio-java-test-2be2djh, requestId=15AC26D496E65E0E, hostId=993c7569-6622-41b6-8994-79246f7c64c2)\nrequest={method=DELETE, url=http://72.28.97.51:32333/minio-java-test-3r6ra4h/minio-java-test-2be2djh?uploadId=8a7e7430-f731-4134-afcf-1aff639b6342, headers=Host: 72.28.97.51:32333\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20190627T200425Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20190627/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 363\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2019-06-27T19-45-26Z\nVary: Origin\nX-Amz-Request-Id: 15AC26D496E65E0E\nX-Xss-Protection: 1; mode=block\nDate: Thu, 27 Jun 2019 20:04:25 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1204), io.minio.MinioClient.execute(MinioClient.java:1070), io.minio.MinioClient.executeDelete(MinioClient.java:1347), io.minio.MinioClient.abortMultipartUpload(MinioClient.java:5149), io.minio.MinioClient.putObject(MinioClient.java:4337), io.minio.MinioClient.putObject(MinioClient.java:4183), FunctionalTest.listIncompleteUploads_test1(FunctionalTest.java:1601), FunctionalTest.runTests(FunctionalTest.java:2988), FunctionalTest.main(FunctionalTest.java:3118)]"
}

Executed 9 out of 14 tests successfully.
```

**Review by krisis — APPROVED** [review]

**Review by krishnasrinivas — APPROVED** [review]

**Pull request merged**